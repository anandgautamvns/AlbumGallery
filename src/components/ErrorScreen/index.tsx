import React, {memo} from 'react';
import {StyleSheet, Text, TouchableOpacity, View} from 'react-native';

interface ErrorScreenProps {
  error: Error | null;
  onRetry?: () => void;
}

const ErrorScreen: React.FC<ErrorScreenProps> = ({error, onRetry}) => {
  const messageError = error?.message || 'Something went wrong.';
  return (
    <View style={styles.container}>
      <Text style={styles.icon}>‚ùå</Text>
      <Text style={styles.title}>Oops!</Text>
      <Text style={styles.message}>{messageError}</Text>
      {onRetry && (
        <TouchableOpacity style={styles.button} onPress={onRetry}>
          <Text style={styles.buttonText}>Retry</Text>
        </TouchableOpacity>
      )}
    </View>
  );
};

export default memo(ErrorScreen);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 24,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#fff',
  },
  icon: {
    fontSize: 64,
    marginBottom: 16,
  },
  title: {
    fontSize: 24,
    fontWeight: '700',
    marginBottom: 8,
    color: '#333',
  },
  message: {
    fontSize: 16,
    textAlign: 'center',
    color: '#666',
    marginBottom: 24,
    paddingHorizontal: 12,
  },
  button: {
    backgroundColor: '#0077c8',
    paddingVertical: 12,
    paddingHorizontal: 32,
    borderRadius: 6,
  },
  buttonText: {
    color: '#fff',
    fontSize: 16,
    fontWeight: '600',
  },
});
