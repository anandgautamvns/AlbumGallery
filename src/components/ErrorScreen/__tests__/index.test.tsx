import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import ErrorScreen from '..';

describe('ErrorScreen Component', () => {
  const mockError = new Error('Test error message');
  const mockOnRetry = jest.fn();

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render with the default error message when no error is provided', () => {
    const {getByText} = render(<ErrorScreen error={null} />);
    expect(getByText('Something went wrong.')).toBeTruthy();
  });

  it('should render with the provided error message', () => {
    const {getByText} = render(<ErrorScreen error={mockError} />);
    expect(getByText('Test error message')).toBeTruthy();
  });

  it('should render the Retry button when onRetry is provided', () => {
    const {getByText} = render(
      <ErrorScreen error={mockError} onRetry={mockOnRetry} />,
    );
    expect(getByText('Retry')).toBeTruthy();
  });

  it('should call onRetry when the Retry button is pressed', () => {
    const {getByText} = render(
      <ErrorScreen error={mockError} onRetry={mockOnRetry} />,
    );
    fireEvent.press(getByText('Retry'));
    expect(mockOnRetry).toHaveBeenCalledTimes(1);
  });

  it('should match the snapshot', () => {
    const {toJSON} = render(
      <ErrorScreen error={mockError} onRetry={mockOnRetry} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
