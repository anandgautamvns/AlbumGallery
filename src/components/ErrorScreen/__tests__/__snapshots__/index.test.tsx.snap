// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorScreen Component should match the snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "padding": 24,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 64,
        "marginBottom": 16,
      }
    }
  >
    ‚ùå
  </Text>
  <Text
    style={
      {
        "color": "#333",
        "fontSize": 24,
        "fontWeight": "700",
        "marginBottom": 8,
      }
    }
  >
    Oops!
  </Text>
  <Text
    style={
      {
        "color": "#666",
        "fontSize": 16,
        "marginBottom": 24,
        "paddingHorizontal": 12,
        "textAlign": "center",
      }
    }
  >
    Test error message
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#0077c8",
        "borderRadius": 6,
        "opacity": 1,
        "paddingHorizontal": 32,
        "paddingVertical": 12,
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 16,
          "fontWeight": "600",
        }
      }
    >
      Retry
    </Text>
  </View>
</View>
`;
